relay-server/src/services/processor.rs

```diff
@@ -1274,23 +1274,41 @@ impl EnvelopeProcessorService {
         event::finalize(state, &self.inner.config)?;
         self.light_normalize_event(state)?;
         dynamic_sampling::normalize(state);
-        event::filter(state)?;
-        dynamic_sampling::run(state, &self.inner.config);
+        event::filter(state, &self.inner.global_config.current())?;
+        // Don't extract metrics if relay can't apply generic inbound filters.
+        // An inbound filter applied in another up-to-date relay in chain may
+        // need to drop the event, and there should not be metrics from dropped
+        // events.
+        // In processing relays, always extract metrics to avoid losing them.
+        let supported_generic_filters = self.inner.global_config.current().filters.is_ok()
+            && relay_filter::are_generic_filters_supported(
+                self.inner
+                    .global_config
+                    .current()
+                    .filters()
+                    .map(|f| f.version),
+                state.project_state.config.filter_settings.generic.version,
+            );
 
-        // Remember sampling decision, before it is reset in `dynamic_sampling::sample_envelope_items`.
-        let sampling_should_drop = state.sampling_result.should_drop();
+        let mut sampling_should_drop = false;
```

Comment:
This doesn't need to be here and also not be mutable.

Just always use `sampling_should_drop = state.sampling_result.should_drop();`

Reply:
Ah nvm sorry I misread the indentation